### ISSUE TITLE
How to adjust the view state for a view on a page? (Drill down dashboard)

### ISSUE THREAD

**Issue by tungs2** [saltcorn/saltcorn#1504]
Hi, 
I have a question regarding generating a view using data from a 1:n table relationship for a drill down dashboard:

I have a page with a view state account=1; There are several views embedded on the page and one view requires a different view state called 'number' which can be found by a table 'Relation' with the fields 'account' and 'no'. 
My question is, how do I pass to the view a state 'number=no'?

I guess it can be done in the View settings as an Extra state Formula, however I cannot figure out the syntax.

Any help appreciated.

Thank you!

**Comment by rbrasier** [comment]
I have also been struggling with 1:n relationships using the Extra State Formula - it feels like there is so much power in this field although it can be tricky how to work out how to harness it. Although yours sounds a bit different to mine, I have also wondered about the syntax, and examples. (Mine is how do i use the extra state formula to filter results based on who a user is 'following' {stored in another joined table}).

Whoever has any suggestions for this question, it would be good if as a community we could to start documenting some common examples that we could provide the wiki or other FAQs.

**Comment by Txikivasco** [comment]
you're right, @rbrasier we need to start documenting some examples to solve common issues. There's so much potential in Saltcorn but also lots of trial and error to get things to work. Hope someone has the time to document their journey through saltcorn to improve the wiki a bit.

**Comment by j2l** [comment]
I'm struggling too. For shared views in the same page, it replaces a user while wanting to change another field related to the LoggedIn user (and not having the right to change user BTW). I simply don't fully get it and I miss examples, simple ones and complex ones.

**Comment by glutamate** [comment]
@j2l what you you trying to do? Maybe we could start a little cookbook/FAQ on this topic

**Comment by Txikivasco** [comment]
Well, for example including views from different tables in the same page. Let's say we have two tables (A and B) and a join table to make N:N relations with both (that makes 3 tables in total). How would you display a "show" view for item 1 and 2 in table A which relates to items 10 11 and 12 in table B (with their respectives show views). 

I think you made a video on the topic in youtube @glutamate but left the N:N part out of the explanation.

**Comment by glutamate** [comment]
@Txikivasco in your example can you clarify: 

Table A has rows with id 1 and 2 (call these A1 and A2)
Table B has rows with id 10 11 and 12. (call these B10, B11 and B12)

there is some N:N relation (say table C)- say A1 related to B10 and B11 and A2 related to B11 and B12. Is this correct?

Now you want to show A1 together with B10 and B11. And this is not just badges, you want to show B10 and B11 in full together with A1 in full - is this correct?

**Comment by Txikivasco** [comment]
Yes, that's correct. Forgive my bad explanation.

**Comment by glutamate** [comment]
@Txikivasco this solution to your problem involves no formulae. I will sketch it here and then as we discussed I want to write these things on the wiki in full.

I will just make it a bit less abstract, otherwise my head will explode. Say we have tables Person (A above; fields name, age etc) and Group (B above; fields name, purpose, budget etc) and PersonParticipatesInGroup (C above; fields person and group, both Keys to their respective tables)

You want to show a page focusing on a single person which shows their name and age, and also a bunch of cards, on for each Group they participate in together with the budget, name and purpose of that group.

you need these views basic views

* ListPerson - not really part of the solution but just a list of persons with name and a link to the ShowPerson view
* ShowPerson - to begin with just put the normal fields, name and age. At the end we will embed a view here
* ShowGroup - just put the fields: name, purpose, budget. 

Now here is the real solution - create these views:

* ShowPersonParticipatesInGroup - Show on PersonParticipatesInGroup. Delete the default fields and embed a view (i.e. in builder, drop in "View" component), view to embed ShowGroup with the relation called something like `group.groups`.
* FeedPersonParticipatesInGroup - A Feed on PersonParticipatesInGroup. The single item view is ShowPersonParticipatesInGroup. Leave all other settings, except "Each in card?" If you really wanted each of the groups to be in its own card.
* Now go to ShowPerson and embed the FeedPersonParticipatesInGroup view, selecting as relation the one called "PersonParticipatesInGroup.person"

**Comment by j2l** [comment]
I thought you were replying to Txikivasco but now realize you were replying to me :smile: 
Thanks @glutamate
